crates/ordinals/src/brc721/collection_id.rs

```diff
@@ -63,6 +63,22 @@ impl Brc721CollectionId {
 			},
 		)
 	}
+
+	pub fn to_leb128(&self) -> Vec<u8> {
```

Comment:
I might be missing something, but it seems that this implementation isn't leveraging LEB128's efficiency as intended. For instance, a `CollectionId { 1, 1 }` currently serializes to **5 bytes**, while my understanding is that it should only take **2 bytes** based on the LEB128 specification.

Could you clarify if there's a specific reason for this behavior? Perhaps I'm overlooking some design consideration.

The following implementation serializes to 2 bytes.
```rust
use std::io;
use leb128::write;

   pub fn serialize<W: io::Write>(&self, mut writer: W) -> io::Result<()> {
        write::unsigned(&mut writer, self.block)?;
        write::unsigned(&mut writer, self.tx as u64)?;
        Ok(())
    }
    ```

Reply:
You're absolutely right about the inefficiency in the current implementation.
This packs both values into a single u128 before encoding. Even for small values like {1,1}, the packed value is positioned far in the number space due to the bit shifting, requiring more bytes.
I've updated the code. Now it encodes each value separately, allowing each to use the minimum number of bytes needed.